### PR TITLE
Feature/mat 4208

### PR DESCRIPTION
**Pull request by RohitKandimalla**
[https://jira.cms.gov/browse/MAT-4208](https://jira.cms.gov/browse/MAT-4208)

1. Takes in a cqlCodesList and validates each code before sending the same object along with validation results.
2. Checks if the code system is available in the mapping document, which is stored in the S3 bucket. If not available it's an invalid CS. also the mapping document is supposed to have FHIR - VSAC version mapping. 
3. Stan and Gail are still working on updating the document, I will reach out to them once again.

### CODE REVIEW

File: src/main/java/cms/gov/madie/terminology/controller/VsacController.java [hunk 1]
```diff
@@ -57,9 +54,12 @@ public ResponseEntity<String> getValueSet(
   }
 
   protected String serializeFhirValueset(ValueSet fhirValueSet) {
+    return parser.encodeResourceToString(fhirValueSet);
+  }
 
-    String serialized = parser.encodeResourceToString(fhirValueSet);
-
-    return serialized;
+  @PutMapping(path = "/validateCodes", produces = MediaType.APPLICATION_JSON_VALUE)
```
Comment (nmorasb) [inline src/main/java/cms/gov/madie/terminology/controller/VsacController.java]: ideally, REST endpoints would be lowercase, hyphenated when multiple words, and avoid verbs. Think of them as resources. For example, /validateCodes could be PUT /validations/codes

Comment (RohitKandimalla) [reply src/main/java/cms/gov/madie/terminology/controller/VsacController.java]: yes make sense, I will update it

File: src/main/java/cms/gov/madie/terminology/webclient/TerminologyServiceWebClient.java [hunk 2]
```diff
@@ -20,43 +22,31 @@ public class TerminologyServiceWebClient {
...
+  public String getServiceTicket(String tgt) {
+    return terminologyClient
+        .post()
+        .uri(String.format(baseUrl + serviceTicketEndpoint, tgt))
+        .contentType(MediaType.APPLICATION_FORM_URLENCODED)
+        .retrieve()
+        .onStatus(HttpStatus::is5xxServerError, ClientResponse::createException)
+        .onStatus(HttpStatus::is4xxClientError, ClientResponse::createException)
+        .bodyToMono(String.class)
+        .block();
```
Comment (sb-cecilialiu) [inline src/main/java/cms/gov/madie/terminology/webclient/TerminologyServiceWebClient.java:49]: Is there a reason why we have to use block here?

**Review by nmorasb — APPROVED** [review]

**Review by gregory-akins — APPROVED** [review]

File: src/main/java/cms/gov/madie/terminology/webclient/TerminologyServiceWebClient.java (hunk 2 above)
Comment (sb-cecilialiu) [reply src/main/java/cms/gov/madie/terminology/webclient/TerminologyServiceWebClient.java:49]: Talked to Rohit, as it's returning stale data so we are making it block as of now. Thanks a lot Rohit! @RohitKandimalla

**Review by sb-cecilialiu — APPROVED** [review]

**Pull request merged**